### PR TITLE
feat(ui): Add custom time range option in time range dropdown

### LINKED ISSUE #11692: Bring over the calendar selection from 1.7.x (no Rome)
as a query builder
with admin
when i select the time rage for my query
then i would like to be able to select an arbitrary start and end time using a calendar

we will need to ensure the range function of the query uses start and stop time if selected

### PR DESCRIPTION
**Pull request by ischolten**
Closes #11692

_Briefly describe your proposed changes:_
Add a '
`Custom Time Range` option to the timerange dropdown that opens a calendar selection to choose the time range
Change the clickOutside component to listen to `mousedown` rather than `click`
Add react-datepicker as dependency
add stop to queries built by the query builder that default to `now()` if a custom time is not selected

  - [x] Rebased/mergeable
  - [x] Tests pass
  - [x] http/swagger.yml updated (if modified Go structs or API)
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/src/shared/components/dateRangePicker/DateRangePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          />
+        </div>
+      </ClickOutside>
+    )
+  }
+
+  private get stylePosition(): CSSProperties {
+    const {position} = this.props
+    const {bottomPosition, topPosition} = this.state
+    const pickerHeight = 410
```
Comment (chnn) [inline ui/src/shared/components/dateRangePicker/DateRangePicker.tsx]: _nit_ Want to extract this to a named screaming snake case constant at the top of the file?

File: ui/src/shared/components/dateRangePicker/DateRangePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+    if (topPosition) {
+      return {
+        top: '14px',
+        right: `${right + 2}px`,
+      }
+    }
+
+    const bottomPx =
+      (bottomPosition || window.innerHeight - top - 15) - pickerHeight / 2
```
Comment (chnn) [inline ui/src/shared/components/dateRangePicker/DateRangePicker.tsx]: What's the `15` for?

File: ui/src/shared/components/dateRangePicker/DateRangePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    return {
+      bottom: `${bottomPx}px`,
+      right: `${right + 2}px`,
```
Comment (chnn) [inline ui/src/shared/components/dateRangePicker/DateRangePicker.tsx]: What's the `2` for?

File: ui/src/shared/components/dateRangePicker/DateRangePicker.tsx (hunk 1 above)
Comment (ischolten) [reply ui/src/shared/components/dateRangePicker/DateRangePicker.tsx]: can do

File: ui/src/shared/components/dateRangePicker/DateRangePicker.tsx (hunk 3 above)
Comment (ischolten) [reply ui/src/shared/components/dateRangePicker/DateRangePicker.tsx]: some additional padding

File: ui/src/shared/components/dateRangePicker/DateRangePicker.tsx (hunk 2 above)
Comment (ischolten) [reply ui/src/shared/components/dateRangePicker/DateRangePicker.tsx]: some additional padding; i can make into a constant

**Pull request merged**